src/WebJobs.Script/Config/ScriptSettingsManager.cs

```diff
@@ -3,19 +3,27 @@
 
 using System;
 using System.Collections.Concurrent;
+using Microsoft.Extensions.Configuration;
 
 namespace Microsoft.Azure.WebJobs.Script.Config
 {
     public class ScriptSettingsManager
     {
         private static ScriptSettingsManager _instance = new ScriptSettingsManager();
         private readonly ConcurrentDictionary<string, string> _settingsCache = new ConcurrentDictionary<string, string>();
+        private Func<IConfiguration> _configurationFactory = BuildConfiguration;
+        private Lazy<IConfiguration> _configuration = new Lazy<IConfiguration>(BuildConfiguration);
```

Comment:
Can we simplify this pattern? Too many layers of indirection here.

Reply:
Had a quick chat with Mike about this.

This is (unfortunately) required right now to enable the scenarios we want to support, but it is also temporary. Created #1959 to document and track the issue.